src/ability/conditions/meetingOrAgenda.ability.ts

```diff
@@ -0,0 +1,68 @@
+import { ForbiddenException, NotFoundException } from "@nestjs/common";
+import { UserReq } from "../../global/types/CustomRequest";
+import prisma from "../../prisma/client";
+import { Agenda, TeamMeeting } from "@prisma/client";
+
+export const manageOwnTeamMeetingOrAgendaById = async ({
+    user,
+    meetingId,
+    agendaId,
+}: {
+    user: UserReq;
+    meetingId?: number;
+    agendaId?: number;
+    subject?: TeamMeeting | Agenda; // If we want to extend some more permissions for any particular subject
+}) => {
+    let meetingOrAgendaTeamId: number;
+    if (meetingId) {
+        const meeting = await prisma.teamMeeting.findUnique({
+            where: {
+                id: meetingId,
+            },
+            select: {
+                voyageTeamId: true,
+            },
+        });
+        if (!meeting) {
+            throw new NotFoundException(
+                `Team Meeting (id:${meetingId}) not found`,
+            );
+        }
+
+        meetingOrAgendaTeamId = meeting.voyageTeamId!;
+    }
+
+    if (agendaId) {
+        const agenda = await prisma.agenda.findUnique({
+            where: {
+                id: agendaId,
+            },
+            select: {
+                teamMeeting: {
+                    select: {
+                        voyageTeamId: true,
+                    },
+                },
+            },
+        });
+        if (!agenda) {
+            throw new NotFoundException(
+                `Team Meeting Agenda (id:${agendaId}) not found`,
+            );
+        }
+
+        meetingOrAgendaTeamId = agenda.teamMeeting.voyageTeamId;
+    }
+
+    if (user.roles?.includes("admin")) return;
```

Comment:
we have to go through the db queries because  in the above queries we are only checking if  a meeting or agenda exist or not for a given id as it is required for all roles.